Azure.MgmtSdk.Analyzers/DataPropertyEtagAnalyzer.cs

```diff
@@ -0,0 +1,82 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System.Collections.Generic;
+using System;
+using System.Collections.Immutable;
+using System.Text.RegularExpressions;
+
+namespace Azure.MgmtSdk.Analyzers
+{
+    /// <summary>
+    /// Analyzer to check resource type 'ResouceType'.
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class DataPropertyEtagAnalyzer : DiagnosticAnalyzer
+    {
+        protected static readonly string Title = "Improper public data property";
+        protected static readonly string MessageFormat = "The data type of a property name '{0}' is '{1}'.";
+        protected static readonly string Description = "Consider to change it to Etag.";
+
+        public const string DiagnosticIdDataPropertyETagName = "AZM0043";
+
+        private static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(DiagnosticIdDataPropertyETagName, Title,
+            MessageFormat, DiagnosticCategory.Naming, DiagnosticSeverity.Warning, isEnabledByDefault: true,
+            description: Description);
+
+        private static readonly Regex suffixRegex = new Regex(".+(?<Suffix>([Ee][Tt]ag))$");
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get { return ImmutableArray.Create(Rule); } }
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+            context.EnableConcurrentExecution();
+            context.RegisterSyntaxNodeAction(SyntaxAnalyzeDataPropertyEtagVariableName, SyntaxKind.VariableDeclaration);
```

Comment:
Will this cause excessive false alerts? (since we're checking variable deceleration)

Reply:
It is not the same. PropertyDeceleration only find the declaration with default settings.
For example:
public string Test { get; set; } will be only detected by PropertyDeceleration
public string Test; will be only detected by VariableDeceleration